portal/src/main/webapp/js/src/oncoprint/setup.js

```diff
@@ -2214,6 +2271,84 @@ window.CreateCBioPortalOncoprintWithToolbar = function (ctr_selector, toolbar_se
 		    State.refreshData();
 		    oncoprint.releaseRendering();
 		});
+		$('#oncoprint_diagram_mutation_color').find('input[type="checkbox"][name="multi_filter"]').change(function (e) {
+			if (e.originalEvent === undefined) {
+				return true;
+			}
+			var new_known_mutation_settings = {
+				recognize_hotspot: $('#oncoprint_diagram_mutation_color').find('input[type="checkbox"][name="hotspots"]').is(":checked"),
+				recognize_cbioportal_count: $('#oncoprint_diagram_mutation_color').find('input[type="checkbox"][name="cbioportal"]').is(":checked"),
+				recognize_cosmic_count: $('#oncoprint_diagram_mutation_color').find('input[type="checkbox"][name="cosmic"]').is(":checked"),
+				recognize_oncokb_oncogenic: $('#oncoprint_diagram_mutation_color').find('input[type="checkbox"][name="oncokb"]').is(":checked"),
+				ignore_unknown: $('#oncoprint_diagram_mutation_color').find('input[type="checkbox"][name="hide_unknown"]').is(":checked"),
+				recognize_binary_filter: $('#oncoprint_diagram_mutation_color').find('input[type="checkbox"][name="binary_filter"]').is(":checked"),
+				recognize_multi_filter: $('#oncoprint_diagram_mutation_color').find('input[type="checkbox"][name="multi_filter"]').is(":checked"),
+			    };
+			if ($('#oncoprint_diagram_mutation_color').find('input[type="checkbox"][name="multi_filter"]').is(":checked")) {
+			    // If multi_filter has just been selected, set all multi value types to true
+			    for (var value in getMultiValuesMap()) {
+				new_known_mutation_settings["recognize_"+value] = true;
+			    }
+			} else if (!$('#oncoprint_diagram_mutation_color').find('input[type="checkbox"][name="multi_filter"]').is(":checked")) {
+				// If multi_filter has just been deselected, set all all multi value types to false
+				for (var value in getMultiValuesMap()) {
+					new_known_mutation_settings["recognize_"+value] = false;
+				}
+			}
+			window.QuerySession.setKnownMutationSettings(new_known_mutation_settings);
+
+			updateMutationColorForm();
+			updateSortByForm();
+			
+			oncoprint.suppressRendering();
+			updateRuleSets();
+			updateSortComparators();
+			State.refreshData();
+			oncoprint.releaseRendering();
```

Comment:
is there no wrapper function that executes all 7 steps above? Are they all needed?